solver/llbop/exec.go

```diff
@@ -102,6 +103,18 @@ func (e *execOp) Run(ctx context.Context, inputs []solver.Ref) ([]solver.Ref, er
 
 		if m.Dest == pb.RootMount {
 			root = mountable
+			readonlyRootFS = m.Readonly
+			if m.Output == pb.SkipOutput && readonlyRootFS {
+				// XXX this duplicates a case from above.
+				active, err := e.cm.New(ctx, ref, cache.WithDescription(fmt.Sprintf("mount %s from exec %s", m.Dest, strings.Join(e.op.Meta.Args, " ")))) // TODO: should be method
```

Comment:
I feel a bit bad duplicating this code here. The comment says `TODO: should be method` but I'm not too sure what the method was intended to be on. Perhaps on `cache.ImmutableRef` (i.e. `ref` here)?

Or maybe just a freestanding helper func (or even a local more `foo := func(...)` closure thing) would be better?